rollup.config.js

```diff
@@ -20,56 +18,27 @@ const PLUGINS = [
     include: ["src/**/*"],
   }),
   json(),
-  replace({
-    "process.env.NODE_ENV": JSON.stringify("production"),
-    preventAssignment: true,
-  }),
   commonJS(),
 ];
 
 export default [
+  // Main entry build (client and shared code)
   {
     input: "src/index.ts",
     external: ["react", "next"],
     output: [
-      { file: pkg.main, format: "cjs", sourcemap: true },
-      { file: pkg.module, format: "es", sourcemap: true },
+      { file: pkg.exports["."].main, format: "cjs", sourcemap: true, banner: "'use client'" },
+      { file: pkg.exports["."].module, format: "es", sourcemap: true, banner: "'use client'" },
     ],
     plugins: PLUGINS,
   },
-  // UMD build with inline PropTypes
+  // Server entry build (server-only code)
   {
-    input: "src/index.ts",
-    external: ["react", "'next/image'"],
+    input: "src/server/index.ts",
     output: [
-      {
-        name: "ImageKitNext",
-        file: pkg.browser,
-        format: "umd",
-        globals: {
-          react: "React",
-          "next/image": "NextImage",
-        },
-        sourcemap: true,
-      },
+      { file: pkg.exports["./server"].main, format: "cjs", sourcemap: true },
+      { file: pkg.exports["./server"].module, format: "es", sourcemap: true },
```

Comment:
In server components, we can add `use server` directive https://nextjs.org/docs/app/api-reference/directives/use-server